File: Zamba/ZambaWeb2013/zamba2010/Zamba.WFActivity.Regular/Action/DoChangeState.vb
Imports Zamba.Data
Imports Zamba.Core
Imports System.Xml.Serialization
''' -----------------------------------------------------------------------------
''' Project	 : Zamba.Business
''' Class	 : Core.DoChangeState
''' 
''' -----------------------------------------------------------------------------
''' <summary>
''' Regla para modificar el estado de una tarea
''' </summary>
''' <remarks>
''' Hereda WFRuleParent
''' </remarks>
''' <history>
''' 	[Hernan]	29/05/2006	Created
''' </history>
''' -----------------------------------------------------------------------------
<RuleCategory("Tareas"), RuleDescription("Cambiar Estado"), RuleHelp("Permite modificar el estado de una tarea al ejecutarse la regla"), RuleFeatures(False)> <Serializable()> _
Public Class DoChangeState
    Inherits WFRuleParent
    Implements IDoChangeState
    Public Overrides Sub Dispose()

    End Sub
    Private _isLoaded As Boolean
    Private _isFull As Boolean
    Private playRule As Zamba.WFExecution.PlayDoChangeState
    Public Overrides ReadOnly Property IsFull() As Boolean
        Get
            Return _isFull
        End Get
    End Property
    Public Overrides ReadOnly Property IsLoaded() As Boolean
        Get
            Return _isLoaded
        End Get
    End Property
    Public Overrides Sub FullLoad()

    End Sub
    Public Overrides Sub Load()

    End Sub
    Public Sub New(ByVal Id As Int64, ByVal Name As String, ByVal wfstepId As Int64, ByVal p_iStateId As Int32, ByVal p_iStepId As Int32)
        MyBase.New(Id, Name, wfstepId) ', ListofRules.DoChangeState)
        m_iStateId = p_iStateId
        m_iStepId = p_iStepId
        Me.playRule = New Zamba.WFExecution.PlayDoChangeState(Me)
    End Sub
    '--ITEMS--
    'StateId=0

    'Properties
    Private m_iStateId As Int32
    Private m_iStepId As Int32

    Public Property StepId() As Int32 Implements IDoChangeState.StepId
        Get
            Return m_iStepId
        End Get
        Set(ByVal value As Int32)
            m_iStepId = value
        End Set
    End Property

    Public Property StateId() As Int32 Implements IDoChangeState.StateId
        Get
            Return m_iStateId
        End Get
        Set(ByVal value As Int32)
            m_iStateId = value
        End Set
    End Property

    '[Ezequiel] 09/06/2009 - Se elimino el Try-Catch
    Public Overloads Overrides Function Play(ByVal results As System.Collections.Generic.List(Of ITaskResult)) As System.Collections.Generic.List(Of ITaskResult)
        Return playRule.Play(results)
    End Function
    Public Overloads Overrides Function Playweb(ByVal results As System.Collections.Generic.List(Of ITaskResult), ByRef RulePendingEvent As RulePendingEvents, ByRef ExecutionResult As RuleExecutionResult, ByRef Params As Hashtable) As System.Collections.Generic.List(Of ITaskResult)
        Return playRule.Play(results)
    End Function
End Class

File: Zamba/zambawindows2013/Zamba2010/Zamba.WorkFlow.SQLExecution/Zamba.WorkFlow.Execution.ActivityLibrary/Activity/DoChangeState.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Reflection;
using System.Diagnostics;
using Zamba.Core;
//using Zamba.WFUserControl;

namespace Zamba.WFActivity.Xoml
{
    /// <summary>
    /// Regla para modificar el estado de una tarea.
    /// </summary>
    [RuleDescription("CAMBIAR ESTADO"), RuleHelp("Modifica el estado de una tarea.")]
    public partial class DoChangeState : ZRule, IResultActivity, IDoChangeState
	{

        private Int32 m_iStateId, m_iStepId;
        private Zamba.WFExecution.PlayDoChangeState play;
        
		public DoChangeState()
		{
			InitializeComponent();
            play = new Zamba.WFExecution.PlayDoChangeState(this);
		}

        protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
        {
            play.Play(Results);
            return ActivityExecutionStatus.Closed;
        }


        #region IDoChangeState Members

        public int StateId
        {
            get
            {
                return m_iStateId;
            }
            set
            {
                m_iStateId = value;
            }
        }

        public string StateName
        {
            get
            {
                return Zamba.Data.WFStepStatesFactory.GetStateNameByStateId(this.m_iStateId);
            }
        }

        public int StepId
        {
            get
            {
                return m_iStepId;
            }
            set
            {
                m_iStepId = value;
            }
        }

        #endregion

        public override void SetParams(object[] Params)
        {
            StateId = Int32.Parse(Params.GetValue(0).ToString());
            StepId = Int32.Parse(Params.GetValue(1).ToString());
        }
        public override System.Collections.Generic.List<object> GetParams()
        {
            System.Collections.Generic.List<object> Params = new System.Collections.Generic.List<object>();
            Params.Add(StateId);
            Params.Add(StepId);
            return Params;
        }
    }
}